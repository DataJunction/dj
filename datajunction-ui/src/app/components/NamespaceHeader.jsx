import { Component } from 'react';
import HorizontalHierarchyIcon from '../icons/HorizontalHierarchyIcon';

export default class NamespaceHeader extends Component {
  render() {
    const { namespace } = this.props;
    const namespaceParts = namespace.split('.');
    const namespaceList = namespaceParts.map((piece, index) => {
      return (
        <li className="breadcrumb-item" key={index}>
          <a
            className="link-body-emphasis"
            href={'/namespaces/' + namespaceParts.slice(0, index + 1).join('.')}
          >
            {piece}
          </a>
        </li>
      );
    });
    return (
      <ol className="breadcrumb breadcrumb-chevron p-3 bg-body-tertiary rounded-3">
        <li className="breadcrumb-item">
          <a href="/">
            <HorizontalHierarchyIcon />
          </a>
        </li>
        {namespaceList}
      </ol>
    );
  }
}
