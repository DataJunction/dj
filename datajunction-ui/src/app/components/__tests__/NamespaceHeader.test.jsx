import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import NamespaceHeader from '../NamespaceHeader';

const renderer = createRenderer();

describe('<NamespaceHeader />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<NamespaceHeader namespace="shared.dimensions.accounts" />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
