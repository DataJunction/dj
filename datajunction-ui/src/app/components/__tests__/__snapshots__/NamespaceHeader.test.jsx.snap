// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NamespaceHeader /> should render and match the snapshot 1`] = `
<ol
  className="breadcrumb breadcrumb-chevron p-3 bg-body-tertiary rounded-3"
>
  <li
    className="breadcrumb-item"
  >
    <a
      href="/"
    >
      <HorizontalHierarchyIcon />
    </a>
  </li>
  <li
    className="breadcrumb-item"
  >
    <a
      className="link-body-emphasis"
      href="/namespaces/shared"
    >
      shared
    </a>
  </li>
  <li
    className="breadcrumb-item"
  >
    <a
      className="link-body-emphasis"
      href="/namespaces/shared.dimensions"
    >
      dimensions
    </a>
  </li>
  <li
    className="breadcrumb-item"
  >
    <a
      className="link-body-emphasis"
      href="/namespaces/shared.dimensions.accounts"
    >
      accounts
    </a>
  </li>
</ol>
`;
